xdsl/dialects/stablehlo.py

```diff
@@ -77,6 +91,88 @@ class PrecisionAttr(EnumAttribute[Precision], SpacedOpaqueSyntaxAttribute):
     name = "stablehlo.precision"
 
 
+@irdl_attr_definition
+class DotAttr(ParametrizedAttribute):
+    """
+    Attribute that models the dimension information for dot.
+
+    https://github.com/openxla/stablehlo/blob/b075e948092d8a27ed0be48f4f8dbaa6df7e2e3e/stablehlo/dialect/StablehloAttrs.td#L82
+    """
+
+    name = "stablehlo.dot"
+
+    lhs_batching_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+    rhs_batching_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+    lhs_contracting_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+    rhs_contracting_dimensions: ParameterDef[ArrayAttr[IntegerAttr[I64]]]
+
+    def print_parameters(self, printer: Printer) -> None:
+        with printer.in_angle_brackets():
+            with printer.indented():
+                printer.print_string("\nlhs_batching_dimensions = [")
+                printer.print_list(
```

Comment:
Does directly printing the `ArrayAttr` not work here?

Reply:
It wants to print the types for each int, I'll look into a more elegant way of doing this